x-pack/plugins/security_solution/public/management/pages/policy/view/policy_forms/protections/ransomware.tsx

```diff
@@ -0,0 +1,364 @@
+      const newPayload = cloneDeep(policyDetailsConfig);
+      for (const os of OSes) {
+        newPayload[os][protection].mode = id;
+        if (id === ProtectionModes.prevent) {
+          newPayload[os].popup[protection].enabled = true;
+        } else {
+          newPayload[os].popup[protection].enabled = false;
+        }
+      }
+      dispatch({
+        type: 'userChangedPolicyConfig',
+        payload: { policyConfig: newPayload },
+      });
+    }
+  }, [dispatch, id, policyDetailsConfig]);
+
+  /**
+   *  Passing an arbitrary id because EuiRadio
+   *  requires an id if label is passed
+   */
+
+  return (
+    <EuiRadio
+      className="policyDetailsProtectionRadio"
+      label={label}
+      id={radioButtonId}
+      checked={selected === id}
+      onChange={handleRadioChange}
+      disabled={selected === ProtectionModes.off}
+    />
+  );
+});
+
+ProtectionRadio.displayName = 'ProtectionRadio';
+
+const SupportedVersionNotice = ({ optionName }: { optionName: string }) => {
+  const version = popupVersionsMap.get(optionName);
+  if (!version) {
+    return null;
+  }
+
+  return (
+    <EuiText color="subdued" size="xs">
+      <i>
+        <FormattedMessage
+          id="xpack.securitySolution.endpoint.policyDetails.supportedVersion"
+          defaultMessage="Agent version {version}"
+          values={{ version }}
+        />
+      </i>
+    </EuiText>
+  );
+};
+
+/** The Ransomware Protections form for policy details
+ *  which will configure for all relevant OSes.
+ */
+export const Ransomware = React.memo(() => {
+  const policyDetailsConfig = usePolicyDetailsSelector(policyConfig);
+  const dispatch = useDispatch();
+  // currently just taking windows.ransomware, but both windows.ransomware and mac.ransomware should be the same value
+  const selected = policyDetailsConfig && policyDetailsConfig.windows.ransomware.mode;
+  const userNotificationSelected =
+    policyDetailsConfig && policyDetailsConfig.windows.popup.ransomware.enabled;
+  const userNotificationMessage =
+    policyDetailsConfig && policyDetailsConfig.windows.popup.ransomware.message;
+
+  const radios: Immutable<
+    Array<{
+      id: ProtectionModes;
+      label: string;
+      protection: 'ransomware';
+    }>
+  > = useMemo(() => {
+    return [
+      {
+        id: ProtectionModes.detect,
+        label: i18n.translate('xpack.securitySolution.endpoint.policy.details.detect', {
+          defaultMessage: 'Detect',
+        }),
+        protection: 'ransomware',
+      },
+      {
+        id: ProtectionModes.prevent,
+        label: i18n.translate('xpack.securitySolution.endpoint.policy.details.prevent', {
+          defaultMessage: 'Prevent',
+        }),
+        protection: 'ransomware',
+      },
+    ];
+  }, []);
+
+  const handleSwitchChange = useCallback(
+    (event) => {
+      if (policyDetailsConfig) {
+        const newPayload = cloneDeep(policyDetailsConfig);
+        if (event.target.checked === false) {
```

Comment:
Surprised you are not getting a type error here (and other areas below), since `event` is not typed.

Reply:
Isn't it 'any'?

Reply:
yeah... event is any. What type error are you expecting?

Reply:
Yeah, just looked at the docs, and it says `any`. I think thats probably not right since we know this is React synthetic event. I guess this is ok.

FYI - I normally give the callback a type that matches where its being used - like:

```
const handleSwitchChange: EuiSwitchProps['onChange'] = useCallback(
```